### PR TITLE
bugfix: yurthub can not exit when SIGINT/SIGTERM happened

### PR DESCRIPTION
**Pull request by rambohe-ch**
/kind bug

#### What this PR does / why we need it:
if you send SIGINT signal(kill -SIGINT pid) to yurthub component, yurthub can not exit and restart.
so i improved the yurthub http/https server by adding graceful stop feature.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: cmd/yurthub/app/start.go [hunk 1]
```diff
@@ -84,30 +81,10 @@ func NewCmdStartYurtHub(stopCh <-chan struct{}) *cobra.Command {
 
 // Run runs the YurtHubConfiguration. This should never exit
 func Run(cfg *config.YurtHubConfiguration, stopCh <-chan struct{}) error {
+	defer cfg.CertManager.Stop()
```
Comment (Congrool) [inline cmd/yurthub/app/start.go:84]: Could we also use stopCh to stop CertManager?

**Comment by Congrool** [comment]
Another question:
Originally, we will create 4 servers: proxy server, secure proxy server, dummyProxyServer and secure dummyProxyServer. 
But in this PR, we will only create 2 or 3 servers: 
- if enableDummyIf is set, there're proxy server, dummyProxyServer and secure dummyProxyServer
- if enableDummyIf is not set, there're proxy server and secure dummyProxyServer

If it was intentional，could you please also give some explanation?

File: cmd/yurthub/app/start.go (hunk 1 above)
Comment (rambohe-ch) [reply cmd/yurthub/app/start.go:84]: no, CertManager have stopCh that create by itself.

File: pkg/yurthub/network/iptables.go [hunk 2]
```diff
@@ -57,14 +57,10 @@ func NewIptablesManager(dummyIfIP, dummyIfPort string) *IptablesManager {
...
-		// accept traffic to 169.254.2.1:10261
+		// accept traffic to 169.254.2.1:10261/169.254.2.1:10268
 		{iptables.Prepend, iptables.TableFilter, iptables.ChainInput, []string{"-p", "tcp", "-m", "comment", "--comment", "for container access hub agent", "--dport", ifPort, "--destination", ifIP, "-j", "ACCEPT"}},
-		// accept traffic from 169.254.2.1:10261
+		// accept traffic from 169.254.2.1:10261/169.254.2.1:10268
 		{iptables.Prepend, iptables.TableFilter, iptables.ChainOutput, []string{"-p", "tcp", "--sport", ifPort, "-s", ifIP, "-j", "ACCEPT"}},
-		// accept traffic to localhost:10261
-		{iptables.Prepend, iptables.TableFilter, iptables.ChainInput, []string{"-p", "tcp", "--dport", ifPort, "--destination", "localhost", "-j", "ACCEPT"}},
-		// accept traffic from localhost:10261
-		{iptables.Prepend, iptables.TableFilter, iptables.ChainOutput, []string{"-p", "tcp", "--sport", ifPort, "-s", "localhost", "-j", "ACCEPT"}},
```
[the remark this answers is not in the document]
Comment (rambohe-ch) [reply pkg/yurthub/network/iptables.go:67]: traffic to localhost can be accepted in default status, so these iptables settings are dummy setting, so remove them.

**Comment by rambohe-ch** [comment]
> Another question: Originally, we will create 4 servers: proxy server, secure proxy server, dummyProxyServer and secure dummyProxyServer. But in this PR, we will only create 2 or 3 servers:
> 
> * if `enableDummyIf` is set, there're proxy server, dummyProxyServer and secure dummyProxyServer
> * if `enableDummyIf` is not set, there're proxy server and secure proxy server
> 
> If it was intentional，could you please also give some explanation?
@Congrool originally, we have two https server: secure proxy server and secure dummyProxyServer, but in real scenario, only one secure server is used. in order to reduce the overhead of Yurthub, so we reduce the unused secure server.

btw: two http server: proxy server and dummyProxyServer(if `enableDummyIf` is true) both are used in real scenario, so we need to keep them unchanged.

**Pull request merged**